src/components/input/Button.js

```diff
@@ -41,9 +41,9 @@ const ButtonNext = function Button({
   return (
     <ButtonBase
       {...htmlAttributes}
-      className={classnames(
-        'focus-visible-ring transition-colors whitespace-nowrap rounded-sm',
-        'flex items-center font-semibold',
+      classes={classnames(
+        // NB: Base classes are applied by ButtonBase
```

Comment:
Here's the abstraction. Button base classes are now applied in `ButtonBase`. This is the maximum amount of abstraction and inheritance I feel comfortable with right now. 🖕🏻